File: POO/learning/src/Exercicios/Ponto.java
package Exercicios;

public class Ponto {
    private float x;
    private float y;

    public float getX() {
        return this.x;
    }

    public float getY() {
        return this.y;
    }

    public void setPonto(float x, float y) {
        this.x = x;
        this.y = y;
    }

    public float distancia(int x, int y) {
        float dist = (float) Math.sqrt(Math.pow(x - this.x, 2) + Math.pow(y - this.y, 2));
        return dist;
    }

    public float distancia(Ponto p) {
        float dist = (float) Math.sqrt(Math.pow(p.getX() - this.x, 2) + Math.pow(p.getY() - this.y, 2));
        return dist;
    }

    public static void main(String[] args) {
        Ponto p = new Ponto(), pontinho = new Ponto();
        pontinho.setPonto(1, 1);
        p.setPonto(0, 0);
        System.out.println(p.distancia(pontinho));
        return;
    }
}

File: 1p/AED1/TAD/ponto.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include "ponto.h"

struct ponto{
  double x, y;
};

struct ponto* Cria_pto(double x, double y) {
  struct ponto *ponto = NULL;
  ponto = (struct ponto*) malloc(sizeof(struct ponto));
  if(ponto != NULL) {
    ponto->x = x;
    ponto->y = y;
  }
  return ponto;
}

void Libera_pto(struct ponto **p) {
  free(*p);
  *p = NULL;

}

double Distancia_pto(struct ponto *p1, struct ponto *p2) {
  if(p1 == NULL || p2 == NULL) {
    return -1; // estou retornando -1 porque a distancia entre dois pontos pode ser 0
  }
  return sqrt(pow(p1->x - p2->x, 2) + pow(p1->y - p2->y, 2));
}